File: Solutions/C#/0011-ContainerWithMostWater/ContainerWithMostWater.cs
using System;

namespace ContainerWithMostWater
{
    public class Solution
    {
        public int MaxArea(int[] height)
        {
            int i = 0;
            int j = height.Length - 1;

            int max = 0;
            while (i < j)
            {
                int currentArea = (j - i) * (height[i] > height[j] ? height[j] : height[i]);
                if (currentArea > max)
                    max = currentArea;

                if (height[i] >= height[j])
                    j--;
                else
                    i++;
            }

            return max;
        }
    }
}

File: Solutions/C++/0011-ContainerWithMostWater/ContainerWithMostWater.cpp
#include <vector>

using namespace std;

class Solution {
public:
    int maxArea(vector<int>& height) {
        int left = 0, right = height.size() - 1, maxLen = 0, current = 0;
        while(left < right) {
            current = (right - left) * min(height[left], height[right]);
            if(current > maxLen)
                maxLen = current;

            if(height[left] < height[right])
                left++;
            else
                right--;
        }

        return maxLen;
    }
};
